examples/jwt_periodic_broadcast_with_expiration.rs

```diff
@@ -0,0 +1,51 @@
+//! How to run this and also generate public_key.der
+//! 1. visit https://jwt.io and select `Algorithm: RS256`
+//! 2. copy the public key into public_key.pm
+//! 3. `openssl rsa -pubin -in public_key.pem -outform DER -out public_key.der -RSAPublicKey_out`
+//! 4. `cargo run --example jwt_periodic_broadcast`
+//! 5. enter browser console (CTRL+SHFT+K)
+//!    run `parseInt((new Date().getTime() + 1 * 60 * 1000)/1000)` and copy the result
+//!    it mean the token will expire in 1 minute
+//! 6. add `exp` field with the previous number in the **PAYLOAD** text field
+//!    For example
+//!    {
+//!        "sub": "1234567890",
+//!        "name": "John Doe",
+//!        "admin": true,
+//!        "iat": 1516239022,
+//!        "exp": 1573596610
+//!    }
+//! 7. copy the token from jwt.io **Encoded** text field
+//! 8. `websocat ws://127.0.0.1:8080/ws/love --header="Authorization: Bearer ${TOKEN}"`
+
+#[global_allocator]
+static GLOBAL: bitwyre_ws_core::mimalloc::MiMalloc = bitwyre_ws_core::mimalloc::MiMalloc;
+
+use bitwyre_ws_core::{init_log, jwt, run_periodic_websocket_service};
+use bitwyre_ws_core::{Auth, AuthHeader, PeriodicWebsocketConfig, PeriodicWebsocketState};
```

Comment:
why is there `Auth` and `AuthHeader` instead of just one object?

Reply:
Auth is an enum and AuthHeader is a struct. I don't think it's a good idea to put it together.

Reply:
Agreed. I just didn't know what the distinction is. If there's confusion in naming like this, it's a sign we need to rename the variables.

Reply:
Also `*WebsocketConfig` struct can be refactored into `WebsocketConfig<T>`